add new method

dll-as opened the pull request:
- Edit messages with support for text, media, and reply markup
- Delete messages from chats or channels with optional revocation
- Project metadata updates: Added homepage and changelog URLs in pyproject.toml

dll-as committed:
feat: add delete_messages method

Implement method to delete messages from chats or channels with options for revocation.

dll-as committed:
feat: add homepage and changelog URLs to project metadata

Include GitHub links for homepage and changelog in pyproject.toml.

dll-as committed:
feat: add edit_message method to Messages class

Implement functionality to edit existing messages with options for text, media, and reply markup.

dll-as committed:
Merge branch 'mivmi:dev' into dev

dll-as committed:
feat: add get_messages method to retrieve messages

Implement method to fetch messages from a chat or channel with detailed type checking and error handling.

mivmi commented on snakegram/core/methods/messages.py:
```diff
@@ -842,6 +846,340 @@ async def forward_messages(
+        input_peer = await self.get_input_peer(target)
+        if isinstance(input_peer, (types.InputPeerChannel, types.InputPeerChat)):
+            request = functions.channels.DeleteMessages(
+                channel= input_peer,
```
The `channel` should be of type `TypeInputChannel`, but here a `TypeInputPeer` is being passed. 
You can convert it using the `helpers.cast_to_input_channel` function.

mivmi commented on snakegram/core/methods/messages.py:
```diff
@@ -842,6 +846,340 @@ async def forward_messages(
+                revoke= revoke
+            )
+
+        return await self._invoke_wait_updates(request, input_peer)
```
You should change this to `return await self(request)`.

mivmi commented on snakegram/core/methods/messages.py:
```diff
@@ -842,6 +846,340 @@ async def forward_messages(
+
+        return await self._invoke_wait_updates(request, input_peer)
+
+    async def edit_message(
```
editing messages should also work for inline messages

mivmi commented on snakegram/core/methods/messages.py:
```diff
@@ -842,6 +846,340 @@ async def forward_messages(
+
+        input_media = None
+        if media is not None:
+            if media == types.InputMediaEmpty():
```
TL objects don't support `==` . e.g
`types.InputMediaEmpty() == types.InputMediaEmpty()` is False
you should use `isinstance`. and there's no need to check `InputMediaEmpty`

dll-as committed:
feat: update message deletion to return affected messages

Change return type to list of affected messages and ensure message integrity across channels.
